src/rules/unit-case/index.js

```diff
@@ -28,11 +29,18 @@ export default function (expectation) {
       const { value } = decl
 
       valueParser(value).walk((node) => {
+        // Remove interpolation characters
+        const value = node.value.replace(/[#@{}]+/g, "")
+        const valueLowerCase = value.toLowerCase()
+
         // Ignore wrong units within `url` function
-        if (node.type === "function" && node.value.toLowerCase() === "url") { return false }
-        if (node.type !== "word" || !isStandardValue(node.value)) { return }
+        if (node.type === "function" && valueLowerCase === "url") { return false }
+        if (node.type !== "word"
+          || !isStandardValue(value)
+          || !isFinite(parseInt(value))
+        ) { return }
 
-        const parsedUnit = valueParser.unit(node.value)
+        const parsedUnit = valueParser.unit(value)
```

Comment:
I mean this whole pattern, ending in `parsedUnit`: Given a value node from the value-parser, analyze it and return either `undefined` or the node's unit.